src/Tribe/Shortcodes/User_Event_Confirmation_List.php

```diff
@@ -80,102 +91,87 @@ protected function generate_attendance_list() {
 	}
 
 	/**
-	 * Queries for events which have attendee posts related to whichever user
-	 * we are interested in.
+	 * Get list of upcoming event IDs for which the specified user is an attendee.
+	 *
+	 * If attending Tribe Events event, only displays upcoming (not yet ended).
+	 * If attending another type of post (e.g. Post or Page), only displays ones where ticket sales are not yet ended.
 	 *
 	 * @return array
 	 */
 	protected function get_upcoming_attendances() {
-		global $wpdb;
-
-		// Use a limit if set to a positive value
-		$limit = $this->params['limit'] > 0
-			? 'LIMIT ' . $wpdb->prepare( '%d', $this->params['limit'] )
-			: '';
-
-		$keys = $this->build_escaped_key_list( $this->get_event_keys() );
-
-		$query = "
-			SELECT ID
-			FROM {$wpdb->postmeta} AS match_user
-
-			JOIN {$wpdb->postmeta} AS match_events
-			  ON match_events.post_id = match_user.post_id
-
-			JOIN {$wpdb->posts} AS event_list
-			  ON match_events.meta_value = ID
-
-			JOIN {$wpdb->postmeta} AS event_end_dates
-			  ON event_end_dates.post_id = ID
-
-			WHERE (
-				-- Match the user
-				match_user.meta_key = '_tribe_tickets_attendee_user_id'
-				AND match_user.meta_value = %d
-			) AND (
-				-- Restrict to upcoming events
-				match_events.meta_key IN ( $keys )
-				AND event_end_dates.meta_key = '_EventEndDateUTC'
-				AND event_end_dates.meta_value > %s
-			)
-			AND NOT EXISTS (
-				SELECT 1
-				  FROM {$wpdb->postmeta} ticket_status
-				 WHERE ticket_status.meta_key = '_wp_trash_meta_status'
-				   AND ticket_status.post_id = match_events.post_id
-			)
-			GROUP BY ID, event_end_dates.meta_value
-			ORDER BY event_end_dates.meta_value
-
-			$limit
-		";
-
-		return (array) $wpdb->get_col( $wpdb->prepare(
-			$query,
-			absint( $this->params['user'] ),
-			current_time( 'mysql' )
-		) );
-	}
+		if ( empty( $this->params['user'] ) ) {
+			return [];
+		}
 
-	/**
-	 * Provides an array containing the value of the ATTENDEE_EVENT_KEY class constant
-	 * for each active ticketing provider.
-	 *
-	 * @return array
-	 */
-	protected function get_event_keys() {
-		$event_keys = array();
-
-		foreach ( Tribe__Tickets__Tickets::modules() as $module_class => $module_instance ) {
-			/**
-			 * The usage of plain `$module_class::ATTENDEE_EVENT_KEY` will throw a `T_PAAMAYIM_NEKUDOTAYIM`
-			 * when using PHP 5.2, which is a fatal.
-			 *
-			 * So we have to construct the constant name using a string and use the `constant` function.
-			 */
-			if ( defined( "$module_class::ATTENDEE_EVENT_KEY" ) ) {
-				$event_keys[] = constant( "$module_class::ATTENDEE_EVENT_KEY" );
-			} else {
-				$event_keys[] = call_user_func( array( $module_class, 'get_key' ), 'ATTENDEE_EVENT_KEY' );
+		$post_ids                    = [];
+		$ticket_ids                  = [];
+		$event_ids                   = [];
+		$upcoming_non_event_post_ids = [];
+
+		// Get all of this user's attendee records
```

Comment:
:no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: a7387f884336516127c0bd819edf4cdc</sub>